litgpt/tokenizer.py

```diff
@@ -97,8 +101,12 @@ def encode(
             bos_id = self.bos_id
             if bos_id is None:
                 raise NotImplementedError("This tokenizer does not have a defined a bos token")
-            tokens = [bos_id] + tokens
-        if eos:
+            if tokens[0] != bos_id:
+                tokens = [bos_id] + tokens
+        if tokens is None:
+            raise ValueError("`tokens` is None")
+
+        if eos and (not tokens or tokens[-1] != self.eos_id):
```

Comment:
Could you add or extend a test to cover this new behavior? Might also be cleaner to do this in a separate PR.

Reply:
Yes, good idea.
I'll make a separate PR.

Reply:
Sounds good @Andrei-Aksionov . I can merge then and you could do that in a separate PR then like you said.